File: src/main/java/io/gittul/infra/auth/jwt/TokenUserInfo.java
package io.gittul.infra.auth.jwt;

import io.gittul.domain.user.entity.Role;
import io.gittul.domain.user.entity.User;

public record TokenUserInfo(
        Long userId,
        String userName,
        String email,
        Role role
) {

    public static TokenUserInfo of(User user) {
        return new TokenUserInfo(
                user.getUserId(),
                user.getUserName(),
                user.getEmail(),
                user.getRole()
        );
    }
}

File: app/src/main/kotlin/io/gittul/app/infra/auth/dto/TokenUserInfo.kt
package io.gittul.app.infra.auth.dto

import io.gittul.core.domain.user.entity.Role
import io.gittul.core.domain.user.entity.User

data class TokenUserInfo(
    val userId: Long,
    val userName: String,
    val email: String,
    val role: Role
) {
    companion object {
        fun of(user: User): TokenUserInfo {
            return TokenUserInfo(
                user.userId,
                user.userName,
                user.email,
                user.role
            )
        }
    }
}
